Documentation/ApiOverview/CachingFramework/Developer/Index.rst

```diff
@@ -1,143 +1,165 @@
+..  note::
+    Since TYPO3v10, you should prefer :ref:`dependency injection
+    <DependencyInjection>` to retrieve cache frontends whenever possible and no
+    longer use the :php:`CacheManager` directly.
 
 Here is some example code which retrieves the cache via dependency injection:
 
+..  code-block:: php
+    :caption: EXT:some_extension/Classes/MyClass.php
 
-.. code-block:: php
-   :caption: EXT:some_extension/Classes/MyClass.php
-
-
-   namespace Vendor\SomeExtension;
+    namespace Vendor\SomeExtension;
 
-   use TYPO3\CMS\Core\Cache\Frontend\FrontendInterface;
+    use TYPO3\CMS\Core\Cache\Frontend\FrontendInterface;
 
-   class MyClass
-   {
-       /**
-        * @var FrontendInterface
-        */
-       private $cache;
+    class MyClass
+    {
+        private FrontendInterface $cache;
 
-       public function __construct(FrontendInterface $cache)
-       {
-           $this->cache = $cache;
-       }
+        public function __construct(FrontendInterface $cache)
+        {
+            $this->cache = $cache;
+        }
 
-       protected function getCachedValue()
-       {
-           $cacheIdentifier = /* ... logic to determine the cache identifier ... */;
+        protected function getCachedValue()
+        {
+            $cacheIdentifier = /* ... logic to determine the cache identifier ... */;
 
-           // If $entry is false, it hasn't been cached. Calculate the value and store it in the cache:
-           if (($value = $this->cache->get($cacheIdentifier)) === false) {
-               $value = /* ... Logic to calculate value ... */;
-               $tags = /* ... Tags for the cache entry ... */
-               $lifetime = /* ... Calculate/Define cache entry lifetime ... */
+            // If $entry is false, it hasn't been cached. Calculate the value and store it in the cache:
+            if (($value = $this->cache->get($cacheIdentifier)) === false) {
+                $value = /* ... Logic to calculate value ... */;
+                $tags = /* ... Tags for the cache entry ... */
+                $lifetime = /* ... Calculate/Define cache entry lifetime ... */
 
-               // Save value in cache
-               $this->cache->set($cacheIdentifier, $value, $tags, $lifetime);
-           }
+                // Save value in cache
+                $this->cache->set($cacheIdentifier, $value, $tags, $lifetime);
+            }
 
-           return $value;
-       }
+            return $value;
+        }
 
-.. tip::
+..  tip::
+    It is not needed to call :php:`$this->cache->has()` before accessing cache
+    entries with :php:`$this->cache->get()` as the latter returns :php:`false`
+    if no entry exists.
 
-   It isn't needed to call :code:`has()` before accessing cache entries with :code:`get()`
-   as the latter returns :code:`false` if no entry exists.
+Since the auto-wiring feature of the dependency injection container cannot
```

Comment:
Maybe link to the according section in the DI chapter?

Reply:
Done